library/std/src/sys/pal/unix/thread.rs

```diff
@@ -318,6 +318,62 @@ impl Drop for Thread {
     }
 }
 
+pub(crate) fn current_os_id() -> Option<u64> {
+    // Most Unix platforms have a way to query an integer ID of the current thread, all with
+    // slightly different spellings.
+    //
+    // The OS thread ID is used rather than `pthread_self` so as to match what will be displayed
+    // for process inspection (debuggers, trace, `top`, etc.).
+    cfg_if::cfg_if! {
+        // Most platforms have a function returning a `pid_t` or int, which is an `i32`.
+        if #[cfg(any(target_os = "android", target_os = "linux"))] {
+            use crate::sys::weak::syscall;
+
+            // `libc::gettid` is only available on glibc 2.30+, but the syscall is available
+            // since Linux 2.4.11.
+            syscall!(fn gettid() -> libc::pid_t;);
```

Comment:
Unrelated, can we modify the macro in another PR to accept "safe extern" functions?

Reply:
Possibly, but there should still be an `unsafe` *somewhere* to promise that the FFI signature is correct.

We don't even really need "safe extern" though, because it's wrapped anyway -- e.g. we could just declare this safe if we wanted:
https://github.com/rust-lang/rust/blob/48aee7e383503c234cce4206dee9f19f57edb617/library/std/src/sys/pal/unix/weak.rs#L176-L186